src/utils/bitfield.rs

```diff
@@ -0,0 +1,231 @@
+use core::ops::{Bound, Range, RangeBounds};
+
+/// Trait for manipulating subsets of integers.
+pub trait BitField {
+    const TYPE_SIZE: usize;
+
+    /// Return a boolean representing the state of the idx'th bit.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the bit, 0 being the rightmost one.
+    ///
+    /// # Panics
+    ///
+    /// This method will panics if the given index is out of range.
+    fn get_bit(self, idx: usize) -> bool;
+
+    /// Return a value representing the state of a range of bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `range` - The range of bits, starting from 0 for the rightmost one.
+    ///
+    /// # Panics
+    ///
+    /// This method will panics if the given range is invalid.
+    fn get_bits<T: RangeBounds<usize>>(self, range: T) -> Self;
```

Comment:
We may want to separate the bit getter from the bit setter with for example, we could then inherit from the bit getter in the bit setter definition:

```rust
pub trait BitGetter {
    // [...]
}

pub trait BitField : BitGetter {
    // [...]
}
```

That way we might specify which type can modify it's own fields and which one can only get some bits. In fact I'm thinking about `VirtualAddress`, we do want to get some bit sometimes to fill some processor structure but we may not want to modify directly the structure's bits.

Reply:
Done in 16ae4c55091fec8209072769b7111929e14b1606